effects/pooterdb/pooterdb.go

```diff
@@ -42,5 +48,144 @@ func (p *Postgres) CreateUser(ctx context.Context, username, password string) (U
 	if err != nil {
 		return "", err
 	}
-	return UserID(strconv.Itoa(id)), nil
+	return types.UserID(strconv.Itoa(id)), nil
+}
+
+func (p *Postgres) FollowUser(ctx context.Context, userID, followID types.UserID) error {
+	u, err := strconv.Atoi(string(userID))
+	if err != nil {
+		return err
+	}
+
+	f, err := strconv.Atoi(string(followID))
+	if err != nil {
+		return err
+	}
+
+	_, err = p.db.Exec(
+		`INSERT INTO followers
+			(id, user_id, follow_id)
+		VALUES
+			(DEFAULT, $1, $2)
+		RETURNING id`, u, f)
+
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (p *Postgres) RetrievePassword(ctx context.Context, userID types.UserID) (string, error) {
+	var password string
+	u, err := strconv.Atoi(string(userID))
+	if err != nil {
+		return password, err
+	}
+
+	result := p.db.QueryRowContext(ctx,
+		`SELECT password FROM users
+		WHERE id = $1`, u)
+
+	err = result.Scan(&password)
+	if err != nil {
+		return "", err
+	}
+	return password, nil
+}
+
+func (p *Postgres) CreatePost(ctx context.Context, userID types.UserID, content string) error {
+	u, err := strconv.Atoi(string(userID))
+	if err != nil {
+		return err
+	}
+
+	_, err = p.db.Exec(
+		`INSERT INTO posts
+			(id, content, user_id, created_at)
+		VALUES
+			(DEFAULT, $1, $2, NOW())
+		RETURNING id`, content, u)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (p *Postgres) ListUserPosts(ctx context.Context, userID types.UserID) ([]types.Post, error) {
+	var posts []types.Post
+	u, err := strconv.Atoi(string(userID))
+	if err != nil {
+		return posts, err
+	}
+	rows, err := p.db.QueryContext(ctx,
+		`SELECT content, user_id, created_at FROM posts
+		WHERE user_id = $1`, u)
+	if err != nil {
+		return posts, err
+	}
+	defer rows.Close()
+	for rows.Next() {
+		var user int64
+		var content string
+		var createdAt time.Time
+		if err := rows.Scan(&content, &user, &createdAt); err != nil {
```

Comment:
i can't because now that we abstracted away usernames/user IDs from customers, i have to convert userids to usernames before putting them into the Post struct.